### PR TITLE
Added DELETE methods to the api.

### PR DESCRIPTION
**Pull request by codezizo**
To note I didn't add it to the command line script, wanted your input on that if you thought it would be good or not?

### CODE REVIEW

**Comment by pieper** [comment]
As far as I know, `delete` is not part of DICOMweb for study/series/instance.

E.e. it's not listed here:
https://www.dicomstandard.org/dicomweb/restful-structure/

Note that google supports proprietary versions here:

https://cloud.google.com/healthcare/docs/dicom#deletion

I totally agree that this is a needed feature, but probably it should be flagged in the docs as a non-standard API.  Ideally we could make this a "draft standard" API to be used in a change proposal.

**Comment by codezizo** [comment]
@pieper I understand that DELETE is not part of the dicomweb standard so I do agree that we should make it as clear as possible to the users. Adding it to the docs is a good suggestion.

File: src/dicomweb_client/api.py [hunk 1]
```diff
@@ -1745,6 +1783,30 @@ def retrieve_study_metadata(
...
+        Parameters
+        ----------
+        study_instance_uid: str
+            unique study identifier
+
+        Returns
+        -------
+        requests.models.Response
+            HTTP response object returned.
+
```
Comment (hackermd) [inline src/dicomweb_client/api.py:1798]: Include `Note` that this is not a standard method:

```suggestion
        Note
        ----
        The Delete Study resource is not part of the DICOM standard
        and may not be supported by all origin servers.
```

File: src/dicomweb_client/api.py [hunk 2]
```diff
@@ -1979,6 +2041,41 @@ def retrieve_series_rendered(
...
+        ) -> None:
+        '''Deletes specified series and its respective instances.
+
+        Parameters
+        ----------
+        study_instance_uid: str
+            unique study identifier
+        series_instance_uid: str
+            unique series identifier
+
```
Comment (hackermd) [inline src/dicomweb_client/api.py:2057]:
```suggestion

        Note
        ----
        The Delete Series resource is not part of the DICOM standard
        and may not be supported by all origin servers.
```

File: src/dicomweb_client/api.py [hunk 3]
```diff
@@ -2145,6 +2242,48 @@ def store_instances(
...
+        series_instance_uid: str
+            unique series identifier
+        sop_instance_uid: str
+            unique instance identifier
+
+        Returns
+        -------
+        requests.models.Response
+            HTTP response object returned.
+
```
Comment (hackermd) [inline src/dicomweb_client/api.py:2266]:
```suggestion

        Note
        ----
        The Delete Instance resource is not part of the DICOM standard
        and may not be supported by all origin servers.
```

File: src/dicomweb_client/api.py [hunk 4]
```diff
@@ -1795,7 +1795,10 @@ def delete_study(self, study_instance_uid: str) -> None:
         -------
         requests.models.Response
             HTTP response object returned.
-
+        Note
+        ----
+        The Delete Study resource is not part of the DICOM standard
+        and may not be supported by all origin servers.
```
Comment (hackermd) [inline src/dicomweb_client/api.py:1801]:
```suggestion
        and may not be supported by all origin servers.
        
```

File: src/dicomweb_client/api.py [hunk 5]
```diff
@@ -2055,6 +2058,10 @@ def delete_series(
         series_instance_uid: str
             unique series identifier
 
+        Note
+        ----
+        The Delete Series resource is not part of the DICOM standard
+        and may not be supported by all origin servers.
```
Comment (hackermd) [inline src/dicomweb_client/api.py:2064]:
```suggestion
        and may not be supported by all origin servers.

```

File: src/dicomweb_client/api.py [hunk 6]
```diff
@@ -2264,6 +2271,10 @@ def delete_instance(
         requests.models.Response
             HTTP response object returned.
 
+        Note
+        ----
+        The Delete Instance resource is not part of the DICOM standard
+        and may not be supported by all origin servers.
```
Comment (hackermd) [inline src/dicomweb_client/api.py:2277]:
```suggestion
        and may not be supported by all origin servers.

```

File: src/dicomweb_client/api.py [hunk 7]
```diff
@@ -1745,6 +1783,34 @@ def retrieve_study_metadata(
...
+        Parameters
+        ----------
+        study_instance_uid: str
+            unique study identifier
+
+        Returns
+        -------
+        requests.models.Response
+            HTTP response object returned.
+        Note
```
Comment (hackermd) [inline src/dicomweb_client/api.py:1798]:
```suggestion

        Note
```

**Comment by codezizo** [comment]
Thanks Markus!

**Comment by codezizo** [comment]
I fixed all flake8 formatting issues and the test passes now. However I seem to be getting unrelated failures... Not sure if this is known and I should ignore it or not.

> FAILED setup.py::FLAKE8 - AttributeError: 'Application' object has no attribute 'parse_preliminary_options_and_args'

**Comment by hackermd** [comment]
> parse_preliminary_options_and_args

I experienced these issues locally on my Macbook at some point as well. However, the tests seem to run fine on TravisCI, so it doesn't seem to be a general issue.

**Comment by hackermd** [comment]
Thanks for your contribution @codezizo. Will draft a new release within the next days and update the package distribution on PyPI.

**Pull request merged**